eidr/schema.go

```diff
@@ -0,0 +1,116 @@
+// This file is part of the go-meta library.
+//
+// Copyright (C) 2017 JAAK MUSIC LTD
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+// If you have any questions please contact yo@jaak.io
+
+package eidr
+
+import (
+	"github.com/meta-network/go-meta/migrate"
+)
+
+// migrations is a set of database migrations to run on a SQLite3 database
+// to prepare it to index EIDR objects
+var migrations = migrate.NewMigrations()
+
+func init() {
+	// migration 1 creates indexes for the following artist properties:
+	//
+	// * Foo - link to spec foo
+	//
+	migrations.Add(1, `
+CREATE TABLE baseobject (
+	doi_id TEXT NOT NULL,
+	structural_type TEXT NOT NULL,
+	referent_type TEXT NOT NULL,
+	status TEXT NOT NULL,
+	resource_name TEXT NOT NULL,
+	resource_name_lang TEXT NOT NULL,
+	resource_name_class TEXT,
+	approximate_length TEXT
```

Comment:
Is it necessary to index on all of these fields? For example, would we want to link based on `resource_name_class` or `approximate_length`?

Reply:
Good point. I'll weed out some obviously irrelevant fields. I would prefer that someone else judges what is relevant or not, though. For example, `resource_name_class` may be relevant, if a name search is for a specific name type only...

Reply:
I think it is fine for someone else to judge what is needed, so right now we should just include the bare minimum because the more fields we index the larger and more time consuming the indexing becomes.